pkg/componentreadiness/jiraautomator/jiraautomator.go

```diff
@@ -0,0 +1,597 @@
+				}
+			} else {
+				description += fmt.Sprintf("{code}%s{code}\n", test.TestName)
+				description += getProbabilityString(test.ReportStatus, *test.FisherExact)
+			}
+		}
+		if linkedIssue != nil {
+			description += fmt.Sprintf("\n h4. Potentially Related Issues:\n")
+			description += fmt.Sprintf("\n* This regression might be related to [%s|%s]. Feel free to link it if found related.\n", linkedIssue.Key, linkedIssue.Self)
+		}
+
+		absUrl, viewUrl, err := j.getComponentReadinessURLsForView(view)
+		if err != nil {
+			return nil, err
+		}
+		description += fmt.Sprintf("\n h4. Useful Component Readiness Links:\n")
+		description += fmt.Sprintf("\nWe are proving the following two links for your convenience:\n")
+		description += fmt.Sprintf("\n- Click [here|%s] to access the component readiness page generated at the time this issue was created.\n", absUrl)
+		description += fmt.Sprintf("\n- Click [here|%s] to access the component readiness page that will be generated at the time when it is clicked. This is useful for developers to verify their fixes.\n", viewUrl)
+
+		summary := fmt.Sprintf("Component Readiness: %s test regressed", component)
+		issue := jira.Issue{
+			Fields: &jira.IssueFields{
+				Description: description,
+				Type: jira.IssueType{
+					Name: "Bug",
+				},
+				Project: jira.Project{
+					Key: jiratype.ProjectKeyOCPBugs,
+				},
+				Components: []*jira.Component{
+					{
+						Name: component,
+					},
+				},
+				Summary: summary,
+				AffectsVersions: []*jira.AffectsVersion{
+					{
+						Name: view.SampleRelease.Release,
+					},
+				},
+				Labels: []string{jiratype.LabelJiraAutomator},
+			},
+		}
+		created, err := j.jiraClient.CreateIssue(&issue)
+		if err != nil {
+			return created, err
+		}
+		// Set Release Blocker field. Jira does not allow setting those during creation. So do it in separate step.
+		return j.updateReleaseBlocker(created, view.SampleRelease.Release)
+	}
+	return nil, nil
+}
+
+func (j JiraAutomator) updateReleaseBlocker(issue *jira.Issue, release string) (*jira.Issue, error) {
+	if j.isPreRelease(release) {
+		unknowns := tcontainer.NewMarshalMap()
+		unknowns[jiratype.CustomFieldReleaseBlockerName] = map[string]string{"value": jiratype.CustomFieldReleaseBlockerValueApproved}
+		issue := jira.Issue{
+			Key: issue.Key,
+			Fields: &jira.IssueFields{
+				Unknowns: unknowns,
+			},
+		}
+		return j.jiraClient.UpdateIssue(&issue)
+	}
+	return issue, nil
+}
+
+// groupRegressedTestsByComponents groups the regressed tests in the report by components.
+// It also takes into consideration a special column grouping. The idea is if a certain column variant (e.g. metal) has more
+// red cells, we will not want to create a jira card for all the components affected. Instead, we will just create
+// one jira card for metal platform.
```

Comment:
I expected this was a "we will ALSO create a card for the metal platform". I expected the regular behavior to continue. What do you think?

Reply:
Do we really want to create for both? I thought the goal was to avoid bombarding teams when there is a column of red cells due to other root cause.